openpype/modules/kitsu/plugins/publish/integrate_kitsu_note.py

```diff
@@ -42,6 +70,15 @@ def process(self, context):
                         "changed!".format(self.note_status_shortname)
                     )
 
+            # If custom comment, create it
+            if self.custom_comment_template["enabled"]:
+                publish_comment = self.safe_format(
```

Comment:
I don't really see the difference between this and what I'm doing currently. Would this function live inside the process function? I rather have it outside to closer go with `A function does one thing`

Reply:
Yes sure, that was my point, replace `safe_format` with this one, and keep it outside process. The idea is that `safe_format` is dedicated to this, therefore I don't think we need a function for it, while `get_publish_comment` makes sense and becomes overridable by giving its own function.

Reply:
I think you're saying that if we change it to for example this:

```python
def format_publish_comment(self, instance):
    """Get publish comment build from template."""
    
    def replace_missing_key(match):
            key = match.group(1)
            if key not in instance.data:
                self.log.warning(
                    "Key "{}" was not found in instance.data "
                    "and will be rendered as "" in the comment".format(key)
                )
                return ""
            else:
                return str(instance.data[key])

    template = self.custom_comment_template["comment_template"]
    pattern = r"\{([^}]*)\}"
    return re.sub(pattern, replace_missing_key, template)
```

The code in process could then just do:
```py
publish_comment = self.format_publish_comment(instance)
```

And as such if someone would like to basically redefine completely how it's formatted they could just inherit and override only `format_publish_comment` without requiring to use the requiring to use the comment template, and potentially even take other things from the instance, like formatting with data from `instance.context` if really needed.

I haven't seen anyone inherit from other Pyblish plug-ins without that being a "base class" meant solely for inheriting from - but I suppose this does open the door to turn it into that.

Reply:
That's what I meant, but it might not be a strong requirement. My point was the `safe_format` function is only used once, so maybe not necessary to be a function, and thinking about a case we would need it to be a function, I proposed this.

Short story: `safe_format` adds confusion IMO and could be more simple.